src/ccontrol/UserQuerySelect.cc

```diff
@@ -538,8 +538,11 @@ void UserQuerySelect::qMetaRegister(std::string const& resultLocation, std::stri
         // Special token #QID# is replaced with query ID later.
         _resultLoc = "table:result_#QID#";
     }
+
+    int chunkCount = _qSession->getChunksSize();
```

Comment:
It's safer to declare it as:
```
int const chunkCount = ...
```

Reply:
I didn't get you. My suggestion was to do this:
```
int const chunkCount = _qSession->getChunksSize();
```
Where:
```
class QuerySession {
public:
    int getChunksSize() const { return _chunks.size(); }
};
```
I don't think this requires any copy constructor because `int` is the built-in primitive data type.